relayer/src/cache.rs

```diff
@@ -110,6 +131,8 @@ impl Cache {
         }
     }
 
+    /// Attempts to fetch the latest [`Height`] value. If a cache miss is encountered,
+    /// this method attempts to insert the height into the cache.
```

Comment:
Should note that technically, a chain's latest height is always increasing. We cache the latest height with a small TTL so that the latest height query returns the same height if the same query is repeated within a small time frame.

We rely on the assumption that the relayer will retry query on the latest height if it needs the chain to progress to the next height. However this assumption may be broken in some part of our code. Notably, the self-connected transfer test fails if the TTL is set to too high. I'll create a separate issue to track this.

Reply:
Submitted #1947.